src/pymodaq/daq_utils/h5exporters.py

```diff
@@ -210,18 +217,160 @@ def _export_data(self, node, filename) -> None:
 
             np.savetxt(filename, data_trans, fmts, '\t', header='#' + '\t'.join(header))
 
+
 @ExporterFactory.register_exporter()
 class H5npyExporter(H5Exporter):
     """ Exporter object for saving nodes as npy files"""
 
     FORMAT_DESCRIPTION = "Binary NumPy format"
     FORMAT_EXTENSION = "npy"
 
-    def _export_data(self, node, filename) -> None:
+    def _export_data(self, node: Node, filename: str) -> None:
         """Export the node as a numpy binary file format"""
+        # String __contain__ method will evaluate to True for CARRAY,EARRAY,VLARRAY,stringARRAY
         if 'ARRAY' in node.attrs['CLASS']:
             data = node.read()
             if not isinstance(data, np.ndarray):
                 data = np.array(data)
 
-            np.save(filename, data)
\ No newline at end of file
+            np.save(filename, data)
+
+
+# This Exporter depends on the availability of the hyperspy package.
```

Comment:
I would place this code into a specific folder. Then modify the factory code to import (and then register) any file into that folder. It means we will easily be able to expand functionalities while not modifying core exporters (text, h5 and npy). The other exporters could also (and later) be installed as pymodaq plugin functionnalities from an external package (as plugins should work). I'm saying this as yours is a bit dedicated and the dependency to hyperspy could in fine be included into the plugin package.  but for the time being placing it into a dedicated folder will do.

Reply:
It can be done, however you need to know that the register Exporter Factory relies on classes being imported otherwise the registration is not performed.  So it's not as simple as moving the code in an other file or folder, but it must also be imported properly somewhere. 

I don't mind but it just adds complexity for something that will be changed afterwards anyway if you want to make export formats a real thing (and honestly, at this point in time I am not aware of any effort to support other file formats. You were thinking of something specific?)

Reply:
Well my goal is to write a developer documentation containing every point where they can add easily functionalities. Such exporters could be it because maybe some users will need text exporters adding some more info into it as the basic version...If we think about it ahead of the need without making it too complex that will help.
As for the importing , I understand. But then when importing your exporter  factory the __init__ file of the module will be loaded. There you can place a mechanism to import all classes with a given signature (inheriting from your base class for instance) in a given custom_exporters (or equivalent name) module.

Reply:
For now I just moved it in a different file and the h5modules.py does the import lifting. In v3 I guess there simply won't be any other file formats added so I guess the module init can wait